### PR TITLE
`dhis2` update readme

### LINKED ISSUE #61: `dhis2` Document which DHIS2 adaptor versions are supported by which OpenFn adaptor versions
Especially with the release of DHIS2 v2.38 (see issue OpenFn/adaptors#62 ), there may be pretty significant differences between DHIS2 API versions. 

On this adaptor repo (and maybe also in the OpenFn app), we should make obvious which DHIS2 API versions are supported by _which_ OpenFn adaptor versions... because some DHIS2 API versions may require a new major release of the OpenFn adaptor. 

At least is it possible to document the DHIS2 API versions the current OpenFn adaptor has been implemented for/tested against?

### PR DESCRIPTION
**Pull request by mtuchi**
## Summary

Update dead links and restructure the readme to be more developer friendly 

Fixes #61 

## Details
- Update DHIS2 API docs link
- Add table of content for easy navigation 
- Add section for unit test and integration test 

## AI Usage

Please disclose how you've used AI in this work (it's cool, we just want to know!):

- [ ] Code generation (copilot but not intellisense)
- [ ] Learning or fact checking
- [ ] Strategy / design
- [ ] Optimisation / refactoring
- [x] Translation / spellchecking / doc gen
- [ ] Other
- [ ] I have not used AI

You can read more details in our [Responsible AI Policy](https://www.openfn.org/ai#pull-request-templates)

## Review Checklist

Before merging, the reviewer should check the following items:

- [x] Does the PR do what it claims to do?
- [ ] If this is a new adaptor, added the adaptor on marketing website ?
- [ ] Are there any unit tests?
- [ ] Is there a changeset associated with this PR? Should there be? Note that
      dev only changes don't need a changeset.
- [x] Have you ticked a box under AI Usage?

### CODE REVIEW

**Comment by mtuchi** [comment]
@josephjclark does this PR need a changeset ?

File: packages/dhis2/README.md [hunk 1]
```diff
@@ -1,98 +1,107 @@
...
+To use this package, execute commands via OpenFn/core from the root of the
+repository:
 
 ```sh
-core execute \
- -l ../language-dhis2 \
- -e ./tmp/expression.js \
- -s ./tmp/state.json \
- -o ./tmp/output.json
+openfn job.js -a dhis2 -s tmp/state.json -o tmp/output.json
```
Comment (josephjclark) [inline packages/dhis2/README.md:28]: The CLI will write to `tmp/output.json` by default so I would exclude it from the command  - keep it shorter and neater

File: packages/dhis2/README.md [hunk 2]
```diff
@@ -1,98 +1,107 @@
...
 
-View the [docs site](https://docs.openfn.org/adaptors/packages/dhis2-docs) for
-full technical documentation.
+## Important Note
 
-There are lots of **examples** for how to use the various helper functions on
-the main
-[Adaptor docs](https://docs.openfn.org/adaptors/packages/dhis2-docs#modules)
-page.
+This adaptor supports **old tracker versions** and is compatible with DHIS2 API
```
Comment (josephjclark) [inline packages/dhis2/README.md:38]: Uh is this right? This is the readme for 6.0.1, which DOES support I think anything from 2.38, but we should say we DON'T support anything earlier

File: packages/dhis2/README.md (hunk 1 above)
Comment (mtuchi) [reply packages/dhis2/README.md]: Updated the command

**Comment by josephjclark** [comment]
Hey @mtuchi - sorry for the slow review. No changeset needed as there's no public changes!

**Pull request merged**